pkg/plugin/api.go

```diff
@@ -76,9 +93,57 @@ func (d *Datasource) getTables(ctx context.Context, database string) ([]string,
 		return nil, 500, err
 	}
 
+	d.cache.Set(key, result, time.Minute*1)
+
 	return result, 0, nil
 }
 
+// getColumns returns a list of column names for the given database and table.
+func (d *Datasource) getColumns(ctx context.Context, database, table string) ([]string, int, error) {
+	key := fmt.Sprintf("columns_%s_%s", database, table)
+	cached, found := d.cache.Get(key)
+	if found {
+		return cached.([]string), 0, nil
+	}
+
+	resp, err := d.executeQuery(ctx, database, fmt.Sprintf(`SELECT SNELLER_DATASHAPE(*) FROM (SELECT * FROM %s LIMIT 1000)`, table))
```

Comment:
`table` is going to need to be SQL-quoted here. It can contain characters like `-` that are operators in SQL.

(The easiest solution is just to format it with `%q` since we support all the usual Go escaping rules inside double-quoted identifiers.)